### PR TITLE
Add a share dataset function

### PR DESCRIPTION
**Pull request by urbanslug**
Allow for the sharing of datasets.

requires https://github.com/onaio/onadata/pull/842

### CODE REVIEW

**Comment by urbanslug** [comment]
Waiting for https://github.com/onaio/onadata/issues/840

**Comment by pld** [comment]
requires https://github.com/onaio/onadata/pull/842

File: src/milia/api/project.cljc [hunk 1]
```diff
@@ -71,14 +71,12 @@
 (defn share
   "Share project with specific user or remove specific user from project"
   ([project-id username role]
-    (share project-id username role false))
+   (share project-id username role false))
   ([project-id username role remove?]
```
Comment (pld) [inline src/milia/api/project.cljc]: change sig to `[project-id username role & [remove?]]` remove other sig above

File: src/milia/api/dataset.cljc [hunk 2]
```diff
@@ -317,3 +317,14 @@
                             [:xform dataset-id])]
     (metadata-files :instance instance-id no-cache?
                     :extra-params extra-params)))
+
+(defn update-xform-meta-permissions
```
Comment (pld) [inline src/milia/api/dataset.cljc]: needs tests

Comment (urbanslug) [inline src/milia/api/dataset.cljc]: Done

File: src/milia/api/project.cljc [hunk 3]
```diff
@@ -70,15 +70,11 @@
...
-  ([project-id username role]
-    (share project-id username role false))
-  ([project-id username role remove?]
-    (let [url (make-url "projects" project-id "share")
-          data {:username username :role role}
-          form-params (if remove?
-                        (merge data {:remove "True"})
-                        data)]
-      (parse-http :put url :http-options {:form-params form-params}))))
+  ([project-id username role & [remove?]]
```
Comment (pld) [inline src/milia/api/project.cljc]: cool, also no more parens around the sig when there's only a single sig, I can make that change now

**Pull request merged**